src/lib/encoder/Decoder.h

```diff
@@ -65,18 +57,29 @@ inline Decoder::Decoder(int ecc_bytes, int color_bits, bool interleave)
  *
  * */
 template <typename STREAM>
-inline unsigned Decoder::do_decode(CimbReader& reader, STREAM& ostream, bool legacy_mode)
+inline unsigned Decoder::do_decode(CimbReader& reader, STREAM& ostream)
 {
-	if (legacy_mode)
+	if (cimbar::Config::legacy_mode())
 		return do_decode_coupled(reader, ostream);
 
-	std::vector<unsigned> interleaveLookup = Interleave::interleave_reverse(reader.num_reads(), _interleaveBlocks, _interleavePartitions);
+	unsigned eccBytes = _useEcc? cimbar::Config::ecc_bytes() : 0;
+	unsigned eccBlockSize = cimbar::Config::ecc_block_size();
+	unsigned colorBits = cimbar::Config::color_bits();
+	unsigned interleaveBlocks = _interleave? cimbar::Config::interleave_blocks() : 0;
+	unsigned interleavePartitions = cimbar::Config::interleave_partitions();
+
+	unsigned bitsPerSymbol = cimbar::Config::symbol_bits();
+	unsigned symCapacity = cimbar::Config::capacity(bitsPerSymbol);
+	unsigned colorCapacity = cimbar::Config::capacity(colorBits);
+	unsigned bitsPerOp = cimbar::Config::bits_per_cell();
+	unsigned fountain_chunks_per_frame = cimbar::Config::fountain_chunks_per_frame(bitsPerOp);
```

Comment:
I'm not sure that we *need* to define these all at the top, old school C style. But we *are* pulling from a config object every time we do this, and it probably helps caching maybe possibly to have it all out of the way up front.

I did do some A/B performance tests, but didn't seem an appreciable difference. Still, I feel better about this way.